File: rust/sdk/src/builders/txn_filter.rs
use crate::types::TxId;

pub const DEFAULT_TXN_LIMIT: u64 = 20;

pub struct TxnFilter<T> {
    pub(crate) filter: T,
    pub(crate) min: TxId,
    pub(crate) max: TxId,
    pub(crate) limit: u64,
}

impl<T> TxnFilter<T> {
    pub fn min_tx(mut self, tx_id: TxId) -> Self {
        self.min = tx_id;
        self
    }

    pub fn max(mut self, tx_id: TxId) -> Self {
        self.max = tx_id;
        self
    }

    pub fn limit(mut self, count: u64) -> Self {
        self.limit = count;
        self
    }
}

File: node/m10-sdk/src/builders/txn_filter.ts
import type { TxId } from "../ids";


export interface ITxnFilter {
    min: TxId;
    max: TxId;
    limit: number;
}

export class TxnFilter {

    private _min: Option<TxId>;
    private _max: Option<TxId>;
    private _limit: Option<number>;

    public static readonly DEFAULT_TXN_LIMIT: number = 20;


    public constructor(properties: ITxnFilter) {
        this._min = properties.min;
        this._max = properties.max;
        this._limit = properties.limit;
    }


    public getMinTxId(): Option<TxId> {
        return this._min;
    }
    public getMaxTxId(): Option<TxId> {
        return this._max;
    }
    public getLimit(): Option<number> {
        return this._limit;
    }

    public min(txId: TxId): TxnFilter {
        this._min = txId;
        return this;
    }

    public max(txId: TxId): TxnFilter {
        this._max = txId;
        return this;
    }

    public limit(count: number): TxnFilter {
        this._limit = count;
        return this;
    }
}
